addon/-private/system/model/internal-model.js

```diff
@@ -885,17 +885,28 @@ export default class InternalModel {
     @method clearRelationships
     @private
   */
-  clearRelationships() {
+  clearRelationships(resetAll = false) {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
-        rel.clear();
-        rel.removeInverseRelationships();
+
+        if (resetAll === true) {
+          rel.clear();
+          rel.removeInverseRelationships();
```

Comment:
We still end up with the name spaghetti with `unloadInternalModelFromInverseFromOwn` and `unloadInternalModelFromInverseFromInverses`